base/path.jl

```diff
@@ -163,17 +163,39 @@ See also: [`basename`](@ref)
 """
     basename(path::AbstractString) -> AbstractString
 
-Get the file name part of a path.
+Get the file/dir name part of a path.
+
+Generally, trailing '/' will be ignored. There're some other exceptions:
+
+* ""  ===> "."
+* "~" ==> basename(ENV["HOME"])
```

Comment:
`homedir()` should be preferred to relying on the existence of an environment variable.